bee-network/bee-autopeering/src/peer/stores.rs

```diff
@@ -0,0 +1,315 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Persistent storage of discovered peers.
+
+use super::{
+    lists::{ActivePeer, ActivePeersList, ReplacementList},
+    peer_id::PeerId,
+    Peer,
+};
+
+use sled::{Batch, Db};
```

Comment:
As said on Discord, I really think we should not use sled directly, but we can revisit later.

Reply:
You probably relate to the discussion about using `bee-storage`. Yea, we can always change, and one nice thing already is, that the whole autopeering crate is at least already generic over the concrete DB being used, so the community could also write their `RocksDb` (or anything other) peer store.

Generally ... I like to build something working with as few as possible dependencies, and then gradually add them if we see need for it. And honestly I am a bit skeptical of dogmatically using `bee-storage` and/or `backstage` or any other crate. I agree it's kinda neat to also use crates in our repo if they are just tangentially related to my problem, but that doesn't mean they always fit 100%. The extensibility of `bee-storage` with the trait per storage operation is very handy for a complex DLT storage, but IMHO overkill for just storing some `PeerId`s, which is why I decided to not go for it. Same with backstage. I don't really felt the need for actors here, because I don't need it's resiliency and its other perks.

But I might be wrong in this case of course, and will switch to any of those when I come to the realisation that I was bloody wrong.

Reply:
> Generally ... I like to build something working with as few as possible dependencies, and then gradually add them if we see need for it.

You are actually bringing a dependency, `sled`, to this crate that wouldn't have been necessary if you used the storage crate because you would only be dealing with a trait, and not a concrete type.

> but IMHO overkill for just storing some PeerId

Having to redo the whole database setup when we already have it was IMO the overkill part. Just adding types to bee-storage is really easy.

> I don't really felt the need for actors here

You do have actors, tasks with a state are actors. Backstage is just a way to manage them, and you wrote a task management module in this PR.

But yeah, I guess it's all preferences :)